src/arrayMethodJoin.js

```diff
@@ -5,7 +5,31 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let delimiter = separator;
+
+    if (delimiter === undefined) {
+      delimiter = ',';
+    }
+
+    if (delimiter === null) {
+      delimiter = 'null';
+    };
+
+    for (let i = 0; i < this.length; i++) {
+      if (((this[i] === null) || (this[i] === undefined))
+       && (i !== (this.length - 1))) {
+        result += String(delimiter);
+      } else if ((i === (this.length - 1)) && (this[i] !== undefined)) {
+        result += this[i];
+      } else if (i !== (this.length - 1)) {
+        result += this[i] + String(delimiter);
+      } else {
+        break;
```

Comment:
It looks too complicated. Consider optimizing your solution without 'else...if'
Add this[i] to the result if it's not null and it's not undefined.
And then add a separator if it's not the last element. That's all